eureka/S5_lightcurve_fitting/fitters.py

```diff
@@ -386,39 +572,56 @@ def dynestyfitter(lc, model, meta, log, **kwargs):
     if nlive < min_nlive:
         log.writelog(f'**** WARNING: You should set run_nlive to at least {min_nlive} ****')
 
-    sampler = NestedSampler(ln_like, ptform, ndims,
+    if hasattr(meta, 'ncpu') and meta.ncpu > 1:
+        pool = Pool(meta.ncpu)
+        queue_size = meta.ncpu
+    else:
+        pool = None
+        queue_size = None
+    sampler = NestedSampler(ln_like, ptform, ndims, pool=pool, queue_size=queue_size,
                             bound=bound, sample=sample, nlive=nlive, logl_args = l_args,
                             ptform_args=[prior1, prior2, priortype])
     sampler.run_nested(dlogz=tol, print_progress=True)  # output progress bar
     res = sampler.results  # get results dictionary from sampler
+    if meta.ncpu > 1:
+        pool.close()
+        pool.join()
 
     logZdynesty = res.logz[-1]  # value of logZ
     logZerrdynesty = res.logzerr[-1]  # estimate of the statistcal uncertainty on logZ
 
-    if meta.run_verbose:
-        log.writelog('')
-        # Need to temporarily redirect output since res.summar() prints rather than returns a string
-        old_stdout = sys.stdout
-        sys.stdout = mystdout = StringIO()
-        res.summary()
-        sys.stdout = old_stdout
-        log.writelog(mystdout.getvalue())
+    log.writelog('', mute=(not meta.run_verbose))
+    # Need to temporarily redirect output since res.summar() prints rather than returns a string
+    old_stdout = sys.stdout
+    sys.stdout = mystdout = StringIO()
+    res.summary()
+    sys.stdout = old_stdout
+    log.writelog(mystdout.getvalue(), mute=(not meta.run_verbose))
 
     # get function that resamples from the nested samples to give sampler with equal weight
     # draw posterior samples
     weights = np.exp(res['logwt'] - res['logz'][-1])
     samples = resample_equal(res.samples, weights)
-    if meta.run_verbose:
-        log.writelog('Number of posterior samples is {}'.format(len(samples)))
+    log.writelog('Number of posterior samples is {}'.format(len(samples)), mute=(not meta.run_verbose))
 
-    medians = []
-    for i in range(len(freenames)):
+    # Compute the medians and uncertainties
+    fit_params = []
+    upper_errs = []
+    lower_errs = []
+    for i in range(ndims):
         q = np.percentile(samples[:, i], [16, 50, 84])
-        medians.append(q[1])
-    fit_params = np.array(medians)
+        lower_errs.append(q[0])
+        fit_params.append(q[1])
+        upper_errs.append(q[2])
+    fit_params = np.array(fit_params)
```

Comment:
We should write the best fit parameters and uncertainties to the log file.  Maybe it's done later on?

Reply:
It is indeed done later (following lines that look like `log.writelog('\nDYNESTY RESULTS:')`)